ccf_transaction_processor/Readme.md

```diff
@@ -3,40 +3,172 @@ Licensed under Creative Commons Attribution 4.0 International License
+
+## Test the Deployment with Ping Test
+
+PDO TP contains a simple `ping` rpc that returns success everytime it is queried. Test the PDO-TP deployment using this rpc. Invoke the following commands to issue 100 ping rpcs. The net througput is reported at the end of the test.
+
+```bash
+cd $PDO_SOURCE_ROOT/ccf_transaction_processor/test
+./test.sh <ip-address>
+```
+
+The CCF port is assumed to be 6006.
+
+## Generate Signing Keys used to sign Payloads of Read Transactions
+
+Responses to read-transactions include a payload signature, where the signature is generated within PDO-TP.
+The required signing keys must be generated before PDO-TP can be opened up for business from PDO clients.
+Invoke the following commands to generate the signing keys.
+
+```bash
+cd $PDO_SOURCE_ROOT/ccf_transaction_processor/generate_signing_keys
+./gen_keys.sh <ip-address>
+```
+
+If successful, the rpc returns after 1) creating one set of signing keys locally within the CCF enclave, and 2) scheduling them for global commit. The corresponding verifying key can be obtained using the `get_ledger_verifying_key`
+rpc. Verifying key is returned only after global commmit is successful. Invoke the following commands to
+check that signing keys have been globally committed before opening up CCF for business from PDO clients.
+
+```bash
+./get_verifying_keys.sh <ip-address>
+```
+
+The read-paylaod-signature feature may be used by PDO clients to establish offline verifiable proof of transaction commits as desired by the PDO smart contract application. Note that for trust purposes, it is recommended that any entity that uses the verifying_key gets it directly from the CCF service using the `get_ledger_verifying_key` rpc.
+
+## Using CCF ledger using PDO
+
+We highlight some quick details about how PDO clients can use a CCF based PDO-TP deployemnt. The information below can be found at [PDO docs](../docs) as well.
+
+1. Set the following environment variables:
+
+```bash
+export PDO_LEDGER_TYPE=ccf
+export PDO_LEDGER_URL=http://ip-address:6006
+```
+
+Here, ip-address is the address of the node which hosts the CCF instance.
+
+2. Set env PDO_LEDGER_KEY_ROOT and save CCF's network certificate `networkcert.pem` and user keys @ PDO_LEDGER_KEY_ROOT. Note that the user cert and private keys must be named as `userccf_cert.pem` and `userccf_privk.pem` respectively.
+
+3. Do a clean build of PDO
+
+```bash
+cd $PDO_SOURCE_ROOT/build/
+make clean
+make
+```
+A clean build is a an easy way to ensure updated creation of config files and PDO keys that are comptabible with CCF.
+
```

Comment:
forcing a clean complete build seems excessive.  It is sufficient to do
```
source ${PDO_SOURCE_ROOT}/build/common-config.sh
make -C ${PDO_SOURCE_ROOT}//build force-conf keys
```
for the redefinition of the exports you have in this section, the `source` is probably also not necessary.  If you run it in hardware mode you will though also have to add the target `register` to the `make` command